index.d.ts

```diff
@@ -0,0 +1,226 @@
+        mkdir(opts: mkdirOpts | string, cb?: (error: err, file: Nullable<MutableFile>) => void): void;
+        upload(opts: uploadOpts | string, source?: BufferString, cb?: uploadCb): Writable;
+        uploadAttribute(type: 0 | 1, data: Buffer, callback?: (error: err, file?: this) => void): void;
+        // Not sure about type of file in delete's cb
+        delete(permanent?: boolean, cb?: (error: err, file?: File) => void): this;
+        moveTo(target: File | string, cb?: (error: err, file?: File) => void): this;
+        setAttributes(attributes: Object, cb?: noop): this;
+        rename(filename: string, cb?: noop): this;
+        setLabel(label: labelType, cb?: noop): this;
+        setFavorite(isFavorite?: boolean, cb?: noop): this;
+        shareFolder(options: linkOpts, cb?: noop): this;
+        // this method has a option paramteter but never uses it, maybe update this in the file or change it here 
+        unshareFolder(cb?: noop): this;
+        importFile(sharedFile: string | File, cb?: (error: err, file?: this) => void): void | this;
+        on(event: 'move', listener: (oldDir: File) => void): this;
+        on(event: 'update', listener: (file: MutableFile) => void): this;
+        on(event: 'delete', listener: (file: Readonly<File>) => void): this;
+        once(event: 'move', listener: (oldDir: File) => void): this;
+        once(event: 'update', listener: (file: MutableFile) => void): this;
+        once(event: 'delete', listener: (file: Readonly<File>) => void): this;
+    }
+    export class AES {
+        key: Buffer;
+        constructor(key: Buffer);
+        encryptCBC(buffer: Buffer): Buffer;
+        decryptCBC(buffer: Buffer): Buffer;
+        stringhash(buffer: Buffer): Buffer;
+        encryptECB(buffer: Buffer): Buffer;
+        decryptECB(buffer: Buffer): Buffer;
+    }
+    // Interfaces & Types
+    type labelType = 0 | 1 | 2 | 3 | 4 | 5 | 6 | 7 | '' | 'red' | 'orange' | 'yellow' | 'green' | 'blue' | 'purple' | 'grey';
+    type StorageStatus = 'ready' | 'connecting' | 'closed';
+    type uploadCb = (error: err, file: MutableFile) => void;
+    type errorCb = (error: err) => void;
+    type BufferString = Buffer | string;
+    type Nullable<T> = T | null;
+    type err = Nullable<Error>;
+    type noop = () => void;
+    type Fetch = any; // Change this if you can get the type of fetch
+    interface StorageOpts {
+        email: string;
+        password: string;
+        autoload?: boolean;
+        autologin?: boolean;
+        keepalive?: boolean;
+    }
+    interface StorageJSON {
+        key: string;
+        sid: string;
+        name: string;
+        user: any; // Not sure what this is
+        options: StorageOpts;
+    }
+    interface APIOpts {
+        gateway?: string;
+        fetch?: Fetch;
+    }
+    interface FileOpts {
+        api?: API;
+        key?: BufferString;
+        directory?: boolean;
+        downloadId: string;
+        loadedFile: string; // Not sure
```

Comment:
File constructor will throw an [error](https://github.com/qgustavor/mega/blob/replacing-request/lib/file.mjs#L15) if `loadedFile` was not in options parameter, so I thought it would make sense to make it required and I just realized I forgot to do the same about the `key` in options

Reply:
It will not throw an error: https://github.com/qgustavor/mega/blob/89e5d05992c73efd2654ffc8433a87bfe9941486/lib/file.mjs#L37-L43

It only throws an error if they are strings and have invalid characters.